RecoTracker/MkFit/plugins/MkFitOutputConverter.cc

```diff
@@ -373,7 +439,26 @@ TrackCandidateCollection MkFitOutputConverter::convertCandidates(const MkFitOutp
         0,                                               // mkFit does not produce loopers, so set nLoops=0
         static_cast<uint8_t>(StopReason::UNINITIALIZED)  // TODO: ignore details of stopping reason as well for now
     );
+
+    chi2.push_back(cand.chi2());
+    states.push_back(tsosDet.first);
   }
+
+  if (algoCandSelection_) {
+    const std::vector<float> dnnScores = computeDNNs(
+        output, states, bs, vertices, session, chi2, mkFitOutput.propagatedToFirstLayer() && doErrorRescale_);
+
+    TrackCandidateCollection reducedOutput;
```

Comment:
```suggestion
    TrackCandidateCollection reducedOutput;
    reducedOutput.reserve(output.size());
```
to avoid unnecessary reallocations